src/main/java/org/springframework/data/couchbase/repository/query/support/N1qlUtils.java

```diff
@@ -145,7 +146,7 @@ public static Expression createWhereFilterForEntity(Expression baseWhereCriteria
                                                       EntityMetadata<?> entityInformation) {
     //add part that filters on type key
     String typeKey = converter.getTypeKey();
-    String typeValue = entityInformation.getJavaType().getName();
+    String typeValue = getTypeValue(entityInformation);
```

Comment:
This wouldn't really work, unless the documents are correctly serialized with the correct type value. Also, this covers only query derivation for n1ql. String based n1ql, spatial, view queries also use type mapping which are not addressed.

Type values are used in the Converter class to serialize entity to Couchbase Document and also in deserialization.